### PR TITLE
QML support

### LINKED ISSUE #1: QML deployment
The last bit that is really missing functionality wise to implement the same functionality linuxdeployqt provides is the deployment of QML files.

I would like to combine both the QML related code in linuxdeployqt (e.g., https://github.com/probonopd/linuxdeployqt/blob/master/tools/linuxdeployqt/shared.cpp#L1513), with the script that @azubieta's Nomad software center uses to bundle its QML dependencies (https://github.com/nomad-desktop/nx-software-center/blob/master/build_scripts/appimage.include.qml.modules.py#L99).

Any help is appreciated. CC @azubieta @probonopd

### PR DESCRIPTION
**Pull request by TheAssassin**
This PR adds support for QML deployment.

This is still WIP, please don't merge until we approve the PR.

Fixes #1.

### CODE REVIEW

**Comment by azubieta** [comment]
This looks quite complete, already test it in https://github.com/linuxdeploy/linuxdeploy-plugin-qt-examples

File: src/qml.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+#pragma once
+
+typedef struct {
+    std::string name;
+    boost::filesystem::path path;
+    boost::filesystem::path relativePath;
+} QmlModuleImport;
+
+static const char* const ENV_KEY_QML_MODULES_PATHS = "LINUXDEPLOY_QT_PLUGIN_QML_MODULES_PATHS";
```
Comment (TheAssassin) [inline src/qml.h]: We should remove the `LINUXDEPLOY_QT_PLUGIN_` prefix here. These env var titles are just way too long.

File: tests/data/qml_project/file.qml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import QtQuick 2.0
+import qml_module 1.0
+Item {
+
+}
```
Comment (TheAssassin) [inline tests/data/qml_project/file.qml]: Missing newline.

File: tests/test_deploy_qml.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+#include <boost/filesystem.hpp>
+#include <gtest/gtest.h>
+
+// local includes
+#include "../src/qml.h"
+#include "../src/util.h"
+
+namespace bf = boost::filesystem;
+
+namespace LINUXDEPLOY_PLUGGIN_QT_TESTS {
```
Comment (TheAssassin) [inline tests/test_deploy_qml.cpp]: We wouldn't really need a namespace here I guess.

Please use a lower-case namespace. I suggest `linuxdeploy::plugin::qt::tests`.

File: src/qml.h (hunk 1 above)
Comment (azubieta) [reply src/qml.h]: I used the prefix to avoid name clashing whit other variables set by the user. But let's hope that it will not happen.

Comment (TheAssassin) [reply src/qml.h]: I am 99.9% sure it won't be a problem.

**Comment by TheAssassin** [comment]
Thanks a lot, @azubieta!

**Pull request merged**